Fix flaky test widgets/serializers/rss_test.py

https://github.com/mitodl/open-discussions/runs/1683500770?check_suite_focus=true


### Steps to Reproduce

This test failed once so it might fail again:

```
_____________ test_url_widget_serialize[6-15-updated_parsed-False] _____________

E       AssertionError: assert {'id': 30, 'w...19:33:50Z'}]}} == {'configuratio... Widget', ...}
E         Omitting 4 identical items, use -vv to show
E         Differing items:
E         {'json': {'entries': [{'description': 'Description 4', 'link': 'http://example.com/4', 'timestamp': '2021-01-11T19:33:... 9', 'link': 'http://example.com/9', 'timestamp': '2021-01-11T19:33:50Z', 'title': 'Title 9'}], 'title': 'RSS Widget'}} != {'json': {'entries': [{'description': 'Description 0', 'link': 'http://example.com/0', 'timestamp': '2021-01-11T19:33:... 5', 'link': 'http://example.com/5', 'timestamp': '2021-01-11T19:33:50Z', 'title': 'Title 5'}], 'title': 'RSS Widget'}}
E         Use -v to get the full diff
```

--- a/widgets/serializers/rss_test.py
+++ b/widgets/serializers/rss_test.py
@@ -18,15 +18,19 @@ def test_url_widget_serialize(
     mocker, raise_exception, timestamp_key, item_count, display_limit
 ):
     """Tests that the rss widget serializes correctly"""
-    entries = [
-        {
-            "title": f"Title {idx}",
-            "description": f"Description {idx}",
-            "link": f"http://example.com/{idx}",
-            timestamp_key: time.gmtime(),
-        }
-        for idx in range(item_count)
-    ]
+    entries = sorted(
+        [
+            {
+                "title": f"Title {idx}",
+                "description": f"Description {idx}",
+                "link": f"http://example.com/{idx}",
+                timestamp_key: time.gmtime(),
+            }
+            for idx in range(item_count)
+        ],
+        reverse=True,
+        key=lambda entry: entry[timestamp_key],
+    )
     mock_parse = mocker.patch("feedparser.parse")
     if raise_exception:
         mock_parse.side_effect = Exception("bad")

